src/focus.service.ts

```diff
@@ -509,6 +509,22 @@ export class FocusService {
   private isFocusable(el: HTMLElement) {
     const role = el.getAttribute('role');
     const tabIndex = el.tabIndex;
+    let visible = el.offsetParent !== null;
+
+    for (let parent = el; parent !== this.root; parent = parent.parentElement) {
+      if (parent.classList.contains('hidden')) {
```

Comment:
That works, but do note that now we have to call getComputedStyle for every parent (until we find one with opacity 0, which is not the common case)